deploy/helm/sumologic/upgrade-1.0.0.sh

```diff
@@ -0,0 +1,302 @@
+fluentd.statefulset.nodeSelector
+fluentd.statefulset.tolerations
+fluentd.statefulset.affinity
+fluentd.statefulset.podAntiAffinity
+fluentd.statefulset.replicaCount
+fluentd.statefulset.resources.limits.memory
+fluentd.statefulset.resources.limits.cpu
+fluentd.statefulset.resources.requests.memory
+fluentd.statefulset.resources.requests.cpu
+fluentd.eventsStatefulset.nodeSelector
+fluentd.eventsStatefulset.tolerations
+fluentd.eventsStatefulset.resources.limits.memory
+fluentd.eventsStatefulset.resources.limits.cpu
+fluentd.eventsStatefulset.resources.requests.memory
+fluentd.eventsStatefulset.resources.requests.cpu"
+
+CLEANUP_CONFIGS="sumologic.events
+sumologic.fluentd
+sumologic.k8sMetadataFilter
+sumologic.kubernetesMeta
+sumologic.kubernetesMetaReduce
+sumologic.addStream
+sumologic.addTime
+deployment
+eventsDeployment"
+
+IFS=$'\n' read -r -d '' -a OLD_CONFIGS <<< "$OLD_CONFIGS"
+IFS=$'\n' read -r -d '' -a NEW_CONFIGS <<< "$NEW_CONFIGS"
+IFS=$'\n' read -r -d '' -a CLEANUP_CONFIGS <<< "$CLEANUP_CONFIGS"
+
+# Override new values.yaml with old configs
+yq m -i -x new.yaml $OLD_VALUES_YAML
+
+# Write values of old configs to renamed configs
+# Then delete old configs from new values.yaml
+for i in ${!OLD_CONFIGS[@]}; do
+  yq w -i new.yaml ${NEW_CONFIGS[$i]} "$(yq r $OLD_VALUES_YAML ${OLD_CONFIGS[$i]})"
+  yq d -i new.yaml ${OLD_CONFIGS[$i]}
+done
+
+# Special case for fluentd.events.WatchResourceEventsOverrides
+# as this config is commented out by default but we will write it as empty string
+# which will not work
+if [ "$(yq r $OLD_VALUES_YAML sumologic.watchResourceEventsOverrides)" = "" ]; then
+  yq d -i new.yaml fluentd.events.watchResourceEventsOverrides
+fi
+
+# Keep image version as 1.0.0
+yq w -i new.yaml image.tag 1.0.0
+
+# Keep pre-upgrade hook
+PRE_UPGRADE="clusterRole
+clusterRoleBinding
+configMap
+job
+serviceAccount"
+IFS=$'\n' read -r -d '' -a PRE_UPGRADE <<< "$PRE_UPGRADE"
+for i in ${!PRE_UPGRADE[@]}; do
+  yq w -i new.yaml sumologic.setup.${PRE_UPGRADE[$i]}.annotations[helm.sh/hook] pre-install,pre-upgrade
+done
+
+# Keep Falco disabled
+yq w -i new.yaml falco.enabled false
+
+# Preserve the functionality of addStream=false or addTime=false
+if [ $(yq r $OLD_VALUES_YAML sumologic.addStream) != "true" ] && [ $(yq r $OLD_VALUES_YAML sumologic.addTime) != "true" ]; then
+  REMOVE="stream,time"
+elif [ $(yq r $OLD_VALUES_YAML sumologic.addStream) != "true" ]; then
+  REMOVE="stream"
+elif [ $(yq r $OLD_VALUES_YAML sumologic.addTime) != "true" ]; then
+  REMOVE="time"
+fi
+
+FILTER="<filter containers.**>
+  @type record_modifier
+  remove_keys $REMOVE
+</filter>"
+
+if [ $(yq r $OLD_VALUES_YAML sumologic.addStream) != "true" ] || [ $(yq r $OLD_VALUES_YAML sumologic.addTime) != "true" ]; then
+  yq w -i new.yaml fluentd.logs.containers.extraFilterPluginConf "$FILTER"
+fi
+
+# Prometheus changes
+# Diff of prometheus regexes:
+# git diff origin/drosiek-lint-values deploy/helm/sumologic/values.yaml | \
```

Comment:
This should be changed into `git diff v1.0.0 ...` /cc @sumo-drosiek

Reply:
No, values.yaml in v1.0.0 is formatted incorrectly, this could be change to:
`git checkout v1.0.0 && cp deploy/helm/sumologic/values.yaml && yq w -i deploy/helm/sumologic/values.yaml image.tag 1.0.0 && git reset --hard && git diff ...`

Reply:
I can do that, do we want to?